pkg/test/vpc_client/bastion.go

```diff
@@ -155,3 +178,121 @@ func (vpc *VPC) DestroyBastionProxy(instance types.Instance) error {
 	}
 	return nil
 }
+
+func generateBcryptPassword(plainPassword string) (string, error) {
+	hashedPassword, err := bcrypt.GenerateFromPassword([]byte(plainPassword), bcrypt.DefaultCost)
+	if err != nil {
+		log.LogError("Generate hashed password failed")
+		return "", nil
+	}
+	return string(hashedPassword), nil
+}
+
+func writePasswordToFile(username, hashedPassword, filePath string) error {
+	_, err := os.Stat(filePath)
+	if os.IsNotExist(err) {
+		dir := path.Dir(filePath)
+		err = os.MkdirAll(dir, os.ModePerm)
+		if err != nil {
+			log.LogError("Create directory failed")
+			return err
+		}
+		_, err = os.Create(filePath)
+		if err != nil {
+			log.LogError("Create passed file failed")
+			return err
+		}
+	} else if err != nil {
+		log.LogError("Stat file failed")
+		return err
+	}
+	file, err := os.OpenFile(filePath, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0600)
+	if err != nil {
+		log.LogError("Open password file failed")
+		return err
+	}
+	defer file.Close()
+
+	line := fmt.Sprintf("%s:%s\n", username, hashedPassword)
+	_, err = file.WriteString(line)
+	if err != nil {
+		log.LogError("Write password file failed")
+		return err
+	}
+	return nil
+}
+
+func loadPrivateKey(privateKeyPath, keypairName string) (ssh.Signer, error) {
+	privateKeyName := fmt.Sprintf("%s/%s-%s", privateKeyPath, keypairName, "keyPair.pem")
+	key, err := os.ReadFile(privateKeyName)
+	if err != nil {
+		log.LogError("Read privte key failed")
+		return nil, err
+	}
+
+	signer, err := ssh.ParsePrivateKey(key)
+	if err != nil {
+		log.LogError("Parse privte key failed")
+		return nil, err
+	}
+	return signer, nil
+}
+
+func uploadFileToBastion(host, port, username, privateKeyPath, keypairName, localFilePath, remoteFilePath string) error {
```

Comment:
There is already a function defined to run ssh command please check with Ying.

Reply:
Updated according to comment.